queries/snakemake/highlights.scm

```diff
@@ -0,0 +1,60 @@
+; inherits: python
+
+;; Compound directives
+["rule" "checkpoint" "module"] @keyword
+
+;; Top level directives (eg. configfile, include)
+(module (directive name: _ @keyword))
+
+;; Subordinate directives (eg. input, output)
+(_ body: (_ (directive name: _ @attribute)))
+
+;; rule/module/checkpoint names
+(rule_definition name: (identifier) @type)
+(module_definition name: (identifier) @type)
+(checkpoint_definition name: (identifier) @type)
+
+
+;; Rule imports
+(rule_import
+  "use" @include
+  "rule" @include
+  "from" @include
+  "as"? @include
+  "with"? @include
+)
+
+
+;; Rule inheritance
+(rule_inheritance
+  "use" @keyword
+  "rule" @keyword
+  "with" @keyword
+)
+
+
+;; Wildcard names
+(wildcard (identifier) @parameter)
+
+
+;; References to directive attributes in wildcard interpolations
+(
+  (identifier) @attribute
+  (#has-ancestor? @attribute "directive")
+  (#has-ancestor? @attribute "block")
+  (#any-of? @attribute
+      "config"
+      "input"
+      "log"
+      "output"
+      "params"
+      "resources"
+      "threads"
+      "wildcards"
+   )
+)
+
+((wildcard (identifier) @attribute)
+  (#any-of? @attribute "config" "input" "log" "output" "params" "resources" "threads" "wildcards"))
+((wildcard (attribute object: (identifier) @attribute))
+  (#any-of? @attribute "config" "input" "log" "output" "params" "resources" "threads" "wildcards"))
```

Comment:
```suggestion
((wildcard (identifier) @variable.builtin)
  (#any-of? @variable.builtin "config" "input" "log" "output" "params" "resources" "threads" "wildcards"))
((wildcard (attribute object: (identifier) @variable.builtin))
  (#any-of? @variable.builtin "config" "input" "log" "output" "params" "resources" "threads" "wildcards"))
```

Reply:
Same for the has-ancestor one, just seeing if that's still needed atm

Reply:
1. I would like to keep the special wildcard identifiers consistent with their counterparts in the directive block and therefore prefer `@attribute` here, if that's possible.

2. But as for `@variable.builtin`: thinking about it, this should probably exist for a select subset of identifiers, one listed example being `config` because this is a global (builtin) variable in the snakemake context. There are a few more of these.
3. I realized `(subscript value: (identifier) @...)` needs to be added here because you can also do `input[0]` in addition to `input.named_something`.

OK on 1/2/3?

Reply:
Ah tbh, for 1) I'm not sure attribute even fits there for the directive names, maybe field or label would be slightly better (I like label more personally)

2) Sure, whichever apply make them `@variable.builtin`

3) sure, add that as well.